backend/Testing/Fixtures/IntegrationFixture.cs

```diff
@@ -0,0 +1,77 @@
+using System.IO.Compression;
+using System.Net;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using LexCore.Utils;
+using Shouldly;
+using Testing.ApiTests;
+using Testing.Services;
+using TusDotNetClient;
+using static Testing.Services.Constants;
+
+namespace Testing.Fixtures;
+
+public class IntegrationFixture : IAsyncLifetime
+{
+    private readonly string _templateRepoName = "test-template-repo.zip";
+    private readonly FileInfo _templateRepoZip = new(Path.Join(BasePath, "_template-repo_.zip"));
+    private readonly DirectoryInfo _templateRepo = new(Path.Join(BasePath, "_template-repo_"));
+    public ApiTestBase AdminApiTester { get; } = new();
+    private string AdminJwt = string.Empty;
+
+    public async Task InitializeAsync()
+    {
+        DeletePreviousTestFiles();
+        Directory.CreateDirectory(BasePath);
+        InitTemplateRepo();
+        AdminJwt = await AdminApiTester.LoginAs(AdminAuth.Username, AdminAuth.Password);
+    }
+
+    public Task DisposeAsync()
+    {
+        return Task.CompletedTask;
+    }
+
+    private static void DeletePreviousTestFiles()
+    {
+        if (Directory.Exists(BasePath)) Directory.Delete(BasePath, true);
+    }
+
+    private void InitTemplateRepo()
+    {
+        var assembly = Assembly.GetExecutingAssembly();
+        var templatePath = Path.Join(Path.GetDirectoryName(assembly.Location), _templateRepoName);
```

Comment:
is this really required? in HgService we can just use a relative path. The reason not to do this is `GetExecutingAssembly` can return different results depending on the environment and it might not actually be what you want if the executing assembly is from your IDE for example, but the tests should always run with the CWD as the location where the content is located, meaning a relative path like we used in the HgService should just work

Reply:
It's not required, I just thought it would give consistent results since a relative path is calculated, IIRC, not relative to the location of the current assembly but relative to the location of the current working directory, which is arbitrary. I tested this against `dotnet test` and found that the *entry* assembly was the dotnet test runner, but the *executing* assembly was Testing.csproj. I didn't test it in IDEs, though; it's possible that some of them handle tests differently.

I could change this to `Assembly.GetAssembly(typeof(IntegrationFixture))` to guarantee that it will always be consistent.

Reply:
Done in commit 900c10b2.

Reply:
Actually, I think the relative path *won't* work. At least for me, because I routinely run `dotnet test` in the root of the repo, but the actual .zip file is stored in `backend/Testing` and copied to the output directory, neither of which is the current working directory when I run `dotnet test`. Now, I could make a habit of running `cd backend/Testing; dotnet test`, but I'd rather use the location of the assembly since that's going to be predictable and consistent, and does not depend on the current working directory.

Reply:
No, it does work. Switched to relative path in commit f032ace2.